diff --git a/dist-cjs/runtimeConfig.js b/dist-cjs/runtimeConfig.js
index 3bc17f643e7bda3082a8beb8c159d6d8ef1d5f8a..258629c7c12d30721167e8dccf8228bb620f9176 100644
--- a/dist-cjs/runtimeConfig.js
+++ b/dist-cjs/runtimeConfig.js
@@ -18,11 +18,11 @@ const smithy_client_1 = require("@smithy/smithy-client");
 const util_defaults_mode_node_1 = require("@smithy/util-defaults-mode-node");
 const smithy_client_2 = require("@smithy/smithy-client");
 const getRuntimeConfig = (config) => {
-    (0, smithy_client_2.emitWarningIfUnsupportedVersion)(process.version);
+    // (0, smithy_client_2.emitWarningIfUnsupportedVersion)(process.version);
     const defaultsMode = (0, util_defaults_mode_node_1.resolveDefaultsModeConfig)(config);
     const defaultConfigProvider = () => defaultsMode().then(smithy_client_1.loadConfigsForDefaultMode);
     const clientSharedValues = (0, runtimeConfig_shared_1.getRuntimeConfig)(config);
-    (0, core_1.emitWarningIfUnsupportedVersion)(process.version);
+    // (0, core_1.emitWarningIfUnsupportedVersion)(process.version);
     return {
         ...clientSharedValues,
         ...config,
diff --git a/dist-es/runtimeConfig.js b/dist-es/runtimeConfig.js
index 4901b93cf794e578ac0c0245c5c216f236e3e62a..ed9fe803f6a77f55c87d980309c120ea70ab423b 100644
--- a/dist-es/runtimeConfig.js
+++ b/dist-es/runtimeConfig.js
@@ -14,11 +14,11 @@ import { loadConfigsForDefaultMode } from "@smithy/smithy-client";
 import { resolveDefaultsModeConfig } from "@smithy/util-defaults-mode-node";
 import { emitWarningIfUnsupportedVersion } from "@smithy/smithy-client";
 export const getRuntimeConfig = (config) => {
-    emitWarningIfUnsupportedVersion(process.version);
+    // emitWarningIfUnsupportedVersion(process.version);
     const defaultsMode = resolveDefaultsModeConfig(config);
     const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
     const clientSharedValues = getSharedRuntimeConfig(config);
-    awsCheckVersion(process.version);
+    // awsCheckVersion(process.version);
     return {
         ...clientSharedValues,
         ...config,
