diff --git a/config/index.js b/config/index.js
index 3003a2ed3ee0920f64e124e95cbfac106b30bb7e..06dd9f3e777c18971f8426af9ab32ad40bb389aa 100644
--- a/config/index.js
+++ b/config/index.js
@@ -90,7 +90,7 @@ export function defineConfig(baseConfig = {}) {
         },
         handler: `${start.appRoot}/entry-server${entryExtension}`,
         middleware: start.middleware,
-        routes: solidStartServerFsRouter({ dir: routeDir, extensions }),
+        routes: [], // solidStartServerFsRouter({ dir: routeDir, extensions }),
         extensions,
         target: "server",
         plugins: async () => {
diff --git a/dist/server/StartServer.jsx b/dist/server/StartServer.jsx
index c7f9a4e5b7620d9673109d7d5ac356aaa6fbe1bf..20bc0ce67541ffa20df7ffe45323447afe2ff55b 100644
--- a/dist/server/StartServer.jsx
+++ b/dist/server/StartServer.jsx
@@ -60,10 +60,8 @@ export function StartServer(props) {
             <HydrationScript />
             {context.assets.map((m) => renderAsset(m))}
           </>} scripts={nonce ? (<>
-              <script nonce={nonce} innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
               <script type="module" nonce={nonce} async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
             </>) : (<>
-              <script innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
               <script type="module" async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
             </>)}>
         {!import.meta.env.START_ISLANDS ? (<Hydration>
diff --git a/dist/server/spa/StartServer.jsx b/dist/server/spa/StartServer.jsx
index e577596037befc8131a59e00a8eef774c8e913e5..27b526ec646a259eccc910c855f5de2ff477b185 100644
--- a/dist/server/spa/StartServer.jsx
+++ b/dist/server/spa/StartServer.jsx
@@ -8,10 +8,8 @@ export function StartServer(props) {
     return (<NoHydration>
       {docType}
       <props.document assets={<>{context.assets.map((m) => renderAsset(m))}</>} scripts={nonce ? (<>
-              <script nonce={nonce} innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
               <script type="module" src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
             </>) : (<>
-              <script innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
               <script type="module" src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
             </>)}/>
     </NoHydration>);
