diff --git a/lib/plugins/fs-watcher.js b/lib/plugins/fs-watcher.js
index 2d4bc9dde00a51c552fc6cf3c8a3f6a18fada000..4cb95ca5f387f784048c667f4d5db75be54d7b3d 100644
--- a/lib/plugins/fs-watcher.js
+++ b/lib/plugins/fs-watcher.js
@@ -54,9 +54,9 @@ export const fileSystemWatcher = () => {
 						});
 					}
 					// debug.hmr("Reload generated pages.");
-					server.ws.send({
-						type: "full-reload",
-					});
+					// server.ws.send({
+					// 	type: "full-reload",
+					// });
 				});
 			}
 		},
